src/common/search/SearchPlays.jsx

```diff
@@ -25,15 +25,14 @@ const SearchPlays = ({ reset, query, onChange }) => {
 
   const handleSearch = (event) => {
```

Comment:
On every key press, 
- this function is invoked, which calls a search API.
- this will put an unnecessary load on the search API. 

Consider **debouncing** the function with a suitable delay (maybe 500ms)

Reply:
Added a separate function `handleChange` as debounce was triggering the `searchText` value on `setSearchText` inside `handleSearch`.